traffic_portal/app/src/common/api/UserService.js

```diff
@@ -17,143 +17,148 @@
  * under the License.
  */
 
-var UserService = function(Restangular, $http, $location, $q, authService, httpService, locationUtils, userModel, messageModel, ENV) {
+var UserService = function($http, locationUtils, userModel, messageModel, ENV) {
 
-    var service = this;
+    const service = this;
 
     this.getCurrentUser = function() {
```

Comment:
getCurrentUser() needs to be implemented like this (or something with similar behavior) so if it fails, the remaining resolves are not attempted at all:

```
    this.getCurrentUser = function() {
        return $http.get(ENV.api['root'] + "user/current").then(
            function(result) {
                userModel.setUser(result.data.response);
                return result.data.response;
            },
            function(err) {
                console.error(err);
                return err;
            }
        );
    };
```

Reply:
TBH I'd consider asynchronous calls being done synchronously much weirder. But the problem is easily fixed - the interceptor will catch it immediately if you throw the error instead of returning it. Either way the promise is "rejected", but it seems like throwing resolves a call stack faster or something.

Regardless, I've got it no longer making other API calls on a page if it hits a 401. I had to decrease the `$timeout` on the redirect, otherwise the messages wouldn't show up. Not sure what that's about.

Reply:
resolves aren't synchronous per se. they are a way to ensure that a state (or a nested state) does not resolve until certain dependencies are fetched. in this case i never want the "private" state (or any state under that) to resolve until the currentUser dependency resolves. you are making a lot of fundamental / architectural changes to this PR that is making me very hesitant  to merge it without a LOT of testing.